keel-web/src/main/kotlin/com/netflix/spinnaker/keel/rest/InteractiveNotificationCallbackController.kt

```diff
@@ -0,0 +1,51 @@
+package com.netflix.spinnaker.keel.rest
+
+import com.netflix.spinnaker.keel.api.ConstraintStatus
+import com.netflix.spinnaker.keel.echo.model.EchoNotification
+import com.netflix.spinnaker.keel.exceptions.InvalidConstraintException
+import com.netflix.spinnaker.keel.persistence.DeliveryConfigRepository
+import com.netflix.spinnaker.keel.yaml.APPLICATION_YAML_VALUE
+import java.time.Instant
+import org.slf4j.LoggerFactory
+import org.springframework.http.MediaType
+import org.springframework.web.bind.annotation.PostMapping
+import org.springframework.web.bind.annotation.RequestBody
+import org.springframework.web.bind.annotation.RequestHeader
+import org.springframework.web.bind.annotation.RequestMapping
+import org.springframework.web.bind.annotation.RestController
+
+@RestController
+@RequestMapping(path = ["/notifications/callback"])
+class InteractiveNotificationCallbackController(
+  private val deliveryConfigRepository: DeliveryConfigRepository
+) {
+  private val log by lazy { LoggerFactory.getLogger(javaClass) }
+
+  @PostMapping(
+    consumes = [MediaType.APPLICATION_JSON_VALUE, APPLICATION_YAML_VALUE],
+    produces = [MediaType.APPLICATION_JSON_VALUE, APPLICATION_YAML_VALUE]
+  )
+  // TODO: This should be validated against write access to a service account. Service accounts should
+  //  become a top-level property of either delivery configs or environments.
```

Comment:
All the pre-work has been done for this! We could make that change at any time.
Would probably also make sense to add an api version for delivery configs around the same time, so we can version them?

Reply:
Yeah, I'm totally in favor of adding an API "header" section to the delivery config! On the service account access check, I seem to recall you mentioned we already had an example of that somewhere?

Reply:
Yep: it's in the `ResourceController` [here](https://github.com/spinnaker/keel/blob/master/keel-web/src/main/kotlin/com/netflix/spinnaker/keel/rest/ResourceController.kt#L102)

Reply:
I think it would be better to resolve this TODO and its companion in `DeliveryConfigController` after `serviceAccount` is moved to the delivery config level.

Reply:
not suggesting the change should be done in this PR, I am just excited that the pre-work is done for this todo!

Reply:
@asher I agree it should be checked in `DeliveryConfigController`, but since this is a separate REST endpoint altogether, don't we need to add that check here as well? In any case, I'd live to leave it as a `TODO` for a future PR. 😄

Reply:
Yes, we need to do the check here. I think the TODO here was actually added after I made a comment to that effect in an earlier thread. Just meant to add that they should be resolved together but not in this PR.